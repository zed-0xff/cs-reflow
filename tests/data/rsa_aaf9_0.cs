	public unsafe static RSACryptoServiceProvider rsa_aaf9(object P_0)
	{
		int num = default(int);
		int num2 = default(int);
		uint num4 = default(uint);
		uint num7 = default(uint);
		int num6 = default(int);
		object obj = default(object);
		object obj2 = default(object);
		while (true)
		{
			if (0)
			{
				goto IL_002c;
			}
			num += 55;
			num2 = 1;
			goto IL_0099;
			IL_009e:
			int num3;
			switch ((num4 = (uint)(num3 ^ -1989972787)) % 6)
			{
			case 0u:
				break;
			case 4u:
				goto IL_0099;
			default:
				goto IL_00c6;
			case 2u:
				goto IL_0124;
			case 5u:
				goto IL_01e3;
			case 3u:
				goto IL_03ac;
			case 1u:
				goto end_IL_0000;
			}
			goto IL_002c;
			IL_002c:
			int num5 = -1471;
			if (true)
			{
				switch ((num7 = (uint)(-((~(~((num6 + -1986340433 * -((-(~(~-733626499)) * -537630607 - -(-(1908805781 * -1828608997 + 2077887227 * 1346880360))) ^ -(-1553494909 * (686367071 + (312079092 + -494026225 + 27057672))))) * -85154045) - (0x7BE84CCB ^ (1409721631 * ~(-109595431572724937)))) ^ ~-1240528765) - ~-790160691))) % 12)
				{
				case 7u:
					break;
				case 11u:
					goto IL_0020;
				case 5u:
					goto IL_0075;
				case 1u:
					goto IL_0124;
				case 4u:
					goto IL_0158;
				default:
					goto IL_02fd;
				case 0u:
					goto IL_0320;
				case 10u:
					goto IL_03ac;
				case 2u:
					goto IL_0414;
				case 3u:
					goto IL_04ba;
				case 8u:
					goto IL_0558;
				case 6u:
					goto IL_0579;
				case 9u:
					goto end_IL_0000;
				}
				goto IL_0014;
			}
			goto IL_0038;
			IL_03ac:
			int num8 = 1773214490;
			goto IL_03b1;
			IL_0579:
			byte[] array;
			if (num2 == 0)
			{
				array = a7fdf43bc042466bb877ac8b21cf6b0a(obj);
				goto IL_0166;
			}
			num8 = 972650391;
			goto IL_03b1;
			IL_03b1:
			num6 = num8;
			num5 = 5791;
			if ((nint)-17648 == -30535)
			{
				goto IL_01e3;
			}
			goto IL_0038;
			IL_0038:
			num3 = -2051510895;
			goto IL_009e;
			IL_0558:
			int num9;
			int num10;
			if (obj2 != null)
			{
				num9 = -1548961433;
				num10 = num9;
			}
			else
			{
				num9 = 1287502044;
				num10 = num9;
			}
			num8 = num9 ^ ((int)num7 * -1302693127);
			goto IL_03b1;
			IL_04ba:
			obj = P_0;
			if ((int)((uint)num2 / 15u / 4) + int.MinValue == -95248 * num2 << 28)
			{
				continue;
			}
			nint num11;
			if ((0x201AD1 ^ (0x24C0 & num7)) != 0)
			{
				int plte = Png_e0d5.PLTE;
				num11 = (((-2779 | (4194304 * plte)) != -2779) ? 1573575832 : 7);
			}
			else
			{
				num11 = -1498565059;
			}
			goto IL_00fa;
			IL_0075:
			return null;
			IL_00fa:
			switch (num11)
			{
			case 5:
				break;
			case 2:
				goto IL_0020;
			case 4:
				goto IL_002c;
			case 1:
				goto IL_0075;
			case 7:
				goto IL_00d0;
			case 3:
				goto IL_0182;
			default:
				goto IL_01e3;
			case 0:
				goto IL_04ba;
			}
			goto IL_0014;
			IL_0414:
			int num12;
			int num13;
			if ((int)/*calli with instance method signature not supportd*/ == 0)
			{
				num12 = -1081067555;
				num13 = num12;
			}
			else
			{
				num12 = -1371548028;
				num13 = num12;
			}
			num8 = num12 ^ ((int)num7 * -697138482);
			goto IL_03b1;
			IL_00c6:
			num11 = 6;
			goto IL_00fa;
			IL_01e3:
			int num14 = -147335856;
			int num15 = num14;
			goto IL_01f1;
			IL_0124:
			if ((int)/*calli with instance method signature not supportd*/ == 0)
			{
				num3 = -2110722696;
				goto IL_009e;
			}
			num14 = 1225091283;
			num15 = num14;
			goto IL_01f1;
			IL_02fd:
			if (((7 * num6 + num6) ^ 0x21CF) == 0 || num5 != -1471)
			{
				goto IL_0158;
			}
			num3 = ((int)num4 * -851159363) ^ -165573862;
			goto IL_009e;
			IL_01f1:
			num8 = num14 ^ ((int)num7 * -447352943);
			goto IL_03b1;
			IL_0099:
			num3 = -724269812;
			goto IL_009e;
			IL_0158:
			array = dc3cf9129ac04d8c8d54c2324583ff0f(obj);
			goto IL_0166;
			IL_0166:
			obj2 = array;
			num8 = -2063688421;
			goto IL_03b1;
			IL_0014:
			num5 = 9420;
			if (true)
			{
				return rsa_a143(obj2);
			}
			goto IL_0020;
			IL_0020:
			num5 = -14005;
			if (0)
			{
				goto IL_002c;
			}
			num2 = 0;
			int num16 = afd1f0100a9e48eb9a6c8975d80c2689;
			if (-67108864 + (int)((uint)num16 / 1024u) == (int)(2048 * (((uint)num16 % 1949u) ^ 0x70EF1C76)))
			{
				continue;
			}
			if (num5 == -14005)
			{
				if (((num4 - 2996) & 1) != (1 & ((0xFF001615u | (1022720 * num4)) >> 8)))
				{
					continue;
				}
				num11 = 3;
				goto IL_00fa;
			}
			goto IL_0320;
			IL_0182:
			num8 = ((int)num7 * -609583405) ^ -1908183904;
			goto IL_03b1;
			IL_00d0:
			num8 = ((int)num7 * -214132724) ^ -328250635;
			goto IL_03b1;
			IL_0320:
			int num17;
			if (num2 == 0)
			{
				num8 = -1949785260;
				num17 = num8;
			}
			else
			{
				num8 = -1551047046;
				num17 = num8;
			}
			goto IL_03b1;
			continue;
			end_IL_0000:
			break;
		}
		return aa966298b6f9473baff9d7a8e3b76841(obj2);
	}
