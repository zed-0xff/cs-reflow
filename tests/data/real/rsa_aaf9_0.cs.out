	public unsafe static RSACryptoServiceProvider rsa_aaf9(object P_0)
	{
	    object obj = default(object);
	    object obj2 = default(object);
	    obj = P_0;
	    if ((int) /*calli with instance method signature not supportd*/ == 0)
	    {
	        byte[] array = dc3cf9129ac04d8c8d54c2324583ff0f(obj);
	        obj2 = array;
	        if (obj2 != null)
	        {
	            return rsa_a143(obj2);
	        }
	    }
	    else if ((int) /*calli with instance method signature not supportd*/ == 0)
	    {
	        byte[] array = dc3cf9129ac04d8c8d54c2324583ff0f(obj);
	        obj2 = array;
	        if (obj2 != null)
	        {
	            return rsa_a143(obj2);
	        }
	    }
	    else
	    {
	        byte[] array = a7fdf43bc042466bb877ac8b21cf6b0a(obj);
	        obj2 = array;
	        if (obj2 != null)
	        {
	            return aa966298b6f9473baff9d7a8e3b76841(obj2);
	        }
	    }
	
	    return null;
	}

